runtest.py

```diff
@@ -69,6 +86,63 @@ def clean_up_output_data(self):
         if os.path.exists(output_dir):
             shutil.rmtree(output_dir)
 
+    def _load_settings(self):
+        try:
+            if os.path.exists(self.firefox_settings):
+                with open(self.firefox_settings) as firefox_settings_fh:
+                    return json.load(firefox_settings_fh)
+        except:
+            return {}
+
+    def _get_firefox_profile(self):
+        self.logger.info('Get Profile: {}'.format(self._firefox_profile_path))
+        return self._firefox_profile_path
+
+    def _create_firefox_profile(self):
+        prefs = self.settings_prefs
+        tmp_profile_dir = tempfile.mkdtemp(prefix='firefoxprofile_')
+        self.logger.info('Creating Profile: {}'.format(tmp_profile_dir))
+        self.logger.info('Profile with prefs: {}'.format(prefs))
+        if sys.platform == 'linux2':
+            firefox_cmd = 'firefox'
+            # the command "--profile <PATH> -silent" doesn't work on Ubuntu
+            os.system('{} --profile {} &'.format(firefox_cmd, tmp_profile_dir))
```

Comment:
you should create empty folder first.

By the way, after the folder be created, it can create profile with `-silent`.
I will update the code.